iceoryx_posh/include/iceoryx_posh/experimental/popo/sample.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_EXPERIMENTAL_POSH_POPO_SAMPLE_HPP
+#define IOX_EXPERIMENTAL_POSH_POPO_SAMPLE_HPP
+
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+#include "iceoryx_utils/cxx/unique_ptr.hpp"
+
+namespace iox
+{
+namespace popo
+{
+
+template <typename T>
+class PublisherInterface;
+
+///
+/// @brief The Sample class is a mutable abstraction over types which are written to loaned shared memory.
+/// These samples are publishable to the iceoryx system.
+///
+template <typename T>
+class Sample
+{
+  public:
+    Sample(cxx::unique_ptr<T>&& samplePtr, PublisherInterface<T>& publisher);
+    /// Creates an empty sample.
+    Sample(std::nullptr_t) noexcept;
+    Sample(const Sample<T>&) = delete;
+    Sample<T>& operator=(const Sample<T>&) = delete;
+    Sample<T>& operator=(Sample<T>&& rhs);
+    Sample(Sample<T>&& rhs);
+    ~Sample();
+
+    ///
+    /// @brief operator =nullptr Clears the sample.
+    /// @return
+    ///
+    Sample<T>& operator=(std::nullptr_t) noexcept;
```

Comment:
I think this assignment operator is actually not necessary: If you do something like this:
```cpp
Sample<int> fuu;
fuu = nullptr; // You already have a conversion operator for nullptr therefore Sample(nullptr) is called and is moved to fuu
```

Reply:
Didn't know the conversion operator could take care of this !